libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -2047,7 +2218,50 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
     }
 
-    private fun createChildScope() = threadController.getMainScopeAndRootJob().scope
+    private fun runOnNavigationLooper(block: () -> Unit) {
+        if (navigationOptions.looper == Looper.myLooper()) {
```

Comment:
Is this check expensive? There are 2 components to it:
1) `Looper.myLooper()` invocation;
2) the equality check itself on loopers.

I'm worried that all the clients who don't need the SDK to work on background thread (and that's nearly all the clients) might get a performance regression for every API call.

Reply:
I run tests on Pixel6 to see potential penalty. 
```
@RunWith(AndroidJUnit4::class)
class ExampleBenchmark {

    @get:Rule
    val benchmarkRule = BenchmarkRule()

    @Test
    fun do_nothing() { // 1.8 ns 
        benchmarkRule.measureRepeated {
        }
    }
    @Test
    fun compare_different_loopers() { // 17.9 ns
        benchmarkRule.measureRepeated {
            Looper.myLooper() == Looper.getMainLooper()
        }
    }

    @Test
    fun compare_same_loopers() { // 26.6 ns  
        benchmarkRule.measureRepeated {
            Looper.getMainLooper() == Looper.getMainLooper()
        }
    }

    @Test
    fun compare_lists() { // 149  ns    
        val list1 = MutableList(10) { it }
        val list2 = MutableList(10) { it }
        benchmarkRule.measureRepeated {
             list1 == list2
        }
    }


}
```
Does the performance penalty seem bad for you?

Reply:
Thanks for the experiment. It looks insignificant compared to other things done in `MapboxNavigation` synchronously.